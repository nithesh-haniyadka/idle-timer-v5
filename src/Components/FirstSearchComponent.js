import React, { useState } from 'react'
import { Button } from 'react-bootstrap';
import { BsSearch } from 'react-icons/bs';
// import Table from 'react-bootstrap/Table';
import {productList} from "../../src/Components/Mockdata/productlistJSON"


import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";

function FirstSearchComponent() {
	
		
	const [products, setProducts] = useState(productList);
	const [searchVal, setSearchVal] = useState("");

	function handleSearchClick() {
		if (searchVal === "") { setProducts(productList); return; }
		const filterBySearch = productList.filter((item) => {
			if (item.toLowerCase()
				.includes(searchVal.toLowerCase())) { return item; }
		})
		setProducts(filterBySearch);
	}
	const mystyle = {
		marginLeft: "600px",
		marginTop: "10px",
		fontWeight: "700"
	};

	return (
		// <div>
		// 	<div style={mystyle}>
		// 		<input onChange={e => setSearchVal(e.target.value)}>
		// 		</input>
		// 		<BsSearch onClick={handleSearchClick} />
		// 	</div>
		// 	<div>

		// 		{products.map((product) => {
		// 			return (
		// 				<div style={mystyle}>{product}</div>
		// 			)
		// 		})
		// 		}

		// 	</div>
		// </div>

		<TableContainer component={Paper}>
     <Table aria-label="simple table">
       <TableHead>
	   <div style={mystyle}>
	   <input placeholder='search' onChange={e => setSearchVal(e.target.value)}></input>
	   <BsSearch onClick={handleSearchClick} />
	   </div>

         <TableRow>
           {/* <TableCell align='center'></TableCell> */}

         </TableRow>
       </TableHead>
       <TableBody align="center">
         {products.map((row) => (
           <TableRow key={row.id}>
             <TableCell component="th" scope="row" align='center'>
               {row}
             </TableCell>

           </TableRow>
         ))}
       </TableBody>
     </Table>
   </TableContainer>

	
	);
}

export default FirstSearchComponent;
