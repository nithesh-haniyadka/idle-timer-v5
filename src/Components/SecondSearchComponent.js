import React, { useState } from 'react'
import { BsSearch } from 'react-icons/bs';

function SecondSearchComponent() {
	const productList = ["Africa"
		, "India"
		, "Australia"
		, "New Zealand"
		, "Russia"
		, "China"];
		
	const [products, setProducts] = useState(productList);
	const [searchVal, setSearchVal] = useState("");
	function handleSearchClick() {
		if (searchVal === "") { setProducts(productList); return; }
		const filterBySearch = productList.filter((item) => {
			if (item.toLowerCase()
				.includes(searchVal.toLowerCase())) { return item; }
		})
		setProducts(filterBySearch);
	}
	const mystyle = {
		marginLeft: "600px",
		marginTop: "10px",
		fontWeight: "700"
	};

	return (
		<div>
			<div style={mystyle}>
				<input onChange={e => setSearchVal(e.target.value)}>
				</input>
				<BsSearch onClick={handleSearchClick} />
			</div>
			<div>

				{products.map((product) => {
					return (
						<div style={mystyle}>{product}</div>
					)
				})
				}

			</div>
		</div>
	);
}

export default SecondSearchComponent;
