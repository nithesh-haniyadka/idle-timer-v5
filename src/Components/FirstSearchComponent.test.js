import { render, fireEvent, screen } from "@testing-library/react";
import FirstSearchComponent from "../../src/Components/FirstSearchComponent";
import SecondSearchComponent from "../Components/SecondSearchComponent";
import LoginNew from "./LoginPageNew";


import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('renders <FirstSearchComponent />', () => {
  it('should pass shallow snapshot test', () => {
    const wrapper = shallow(<FirstSearchComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders SecondSearchComponent', () => {
    const wrapper = shallow(<SecondSearchComponent />);
    expect(wrapper).toMatchSnapshot();
  });




});

//test block
test("Test FirstSearchComponent", () => {
// render the component on virtual dom
render(<FirstSearchComponent />);



// //select the elements you want to interact with
// // const counter = screen.getByTestId("counter");
// // const incrementBtn = screen.getByTestId("increment");

// // //interact with those elements
// // fireEvent.click(incrementBtn);

// // //assert the expected result
// // expect(counter).toHaveTextContent("1");
});