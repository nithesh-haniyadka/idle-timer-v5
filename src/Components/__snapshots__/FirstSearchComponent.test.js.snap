// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> should pass shallow snapshot test 1`] = `ShallowWrapper {}`;

exports[`renders <FirstSearchComponent /> renders LoginPage 1`] = `ShallowWrapper {}`;

exports[`renders <FirstSearchComponent /> renders SecondSearchComponent 1`] = `ShallowWrapper {}`;

exports[`renders <FirstSearchComponent /> should pass shallow snapshot test 1`] = `ShallowWrapper {}`;
